### PR TITLE
Add initial functions

### PR DESCRIPTION
**Pull request by paulcsmith**
* build/2
* create/2
* create_pair/2
* create_list/2
* assoc/2

### CODE REVIEW

**Comment by jsteiner** [comment]
What do you think about making separate test files for `build`, `create`, etc?

File: test/anvil_test.exs [hunk 1]
```diff
@@ -1,7 +1,131 @@
...
+  end
+
+  test "create/2 builds the fixture and saves it in the repo" do
+    MyApp.Anvil.create(:user, admin: true)
+
+    assert_received {:created, %{name: "John Doe", admin: true}}
+  end
+
+  test "create_pair/2 creates the fixture and saves it 2 times" do
+    MyApp.Anvil.create_pair(:user, admin: true)
```
Comment (jsteiner) [inline test/anvil_test.exs]: should we assert that this actually returns a pair?

Comment (jsteiner) [inline test/anvil_test.exs]: same with `create` and `create_list`

Comment (paulcsmith) [inline test/anvil_test.exs]: Oh yes of course! Good catch

File: test/anvil_test.exs [hunk 2]
```diff
@@ -1,7 +1,131 @@
...
-  test "the truth" do
-    assert 1 + 1 == 2
+  defmodule TestRepo do
+    def insert!(record) do
+      send self, {:created, record}
+      record
+    end
+  end
+
+
```
Comment (jsteiner) [inline test/anvil_test.exs]: :space_invader:

File: test/anvil_test.exs [hunk 3]
```diff
@@ -1,7 +1,131 @@
...
+  defmodule MyApp.Anvil do
+    use Anvil
+
+    def fixture(:without_opts) do
```
Comment (jsteiner) [inline test/anvil_test.exs]: Aren't these more `factories` than `fixtures`?

Comment (paulcsmith) [inline test/anvil_test.exs]: I feel like this function is a fixture. The other functions are factory functions to do something with the fixture. Maybe that's wrong though :S

File: test/anvil_test.exs (hunk 2 above)
Comment (paulcsmith) [inline test/anvil_test.exs]: Good catch

File: lib/anvil.ex [hunk 4]
```diff
@@ -1,2 +1,63 @@
...
+      message = "No fixture defined for #{inspect fixture_name}"
+      %UndefinedFixture{message: message}
+    end
+  end
+
+  defmacro __using__(_opts) do
+    quote do
+      @before_compile unquote(__MODULE__)
+
+      def assoc(opts, fixture_name) do
```
Comment (BlakeWilliams) [inline lib/anvil.ex]: Thoughts on putting these methods outsude if `__using__` or a separate module so you can document them? https://github.com/BlakeWilliams/Elixir-Slack/blob/master/lib/slack.ex#L75

File: test/anvil_test.exs (hunk 2 above)
Comment (BlakeWilliams) [inline test/anvil_test.exs]: Extra newline

File: test/anvil_test.exs (hunk 3 above)
Comment (paulcsmith) [inline test/anvil_test.exs]: Actually, maybe factory makes the most sense here. I'll think about it a bit more

Comment (paulcsmith) [inline test/anvil_test.exs]: @BlakeWilliams Any thoughts on `factory` versus `fixture` here?

File: lib/anvil.ex (hunk 4 above)
Comment (paulcsmith) [inline lib/anvil.ex]: I think that is a great idea, but I think the problem is that it doesn't actually define the functions. It only import them. We need to define the functions so you can call `create`, `build`, etc. 

Instead I can define the function in the using macro and then call function in `Anvil`. I think that is a good balance between the two

**Pull request merged**